public/assets/ts/admin/editor.ts

```diff
@@ -4,28 +4,48 @@ class Editor {
   editor: any;
   sourceDataURL: string;
   name: string;
+  rtEditor: any;
 
   constructor(
     container: HTMLElement,
     sourceDataURL: string,
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    options?: {
+    options: {
       name?: string;
+      onEditable: (node) => void;
+      onEvent: ({ field }, event: Event) => void;
     }
   ) {
-    this.options = {};
+    this.rtEditor = new window["Quill"]("#quill-editor", { theme: "snow" });
+    this.options = {
+      onEditable: this.onEditorEditable,
+      onEvent: this.onEditorEvent.bind(this),
+    };
+
     this.sourceDataURL = sourceDataURL;
     this.editor = new window["JSONEditor"](container, this.options);
 
     this.name = this.getFormattedURL(this.sourceDataURL.split("path=")[1]);
-    this.createSaveButton(container);
+    this.createButtons(container);
 
     (async () => {
       this.editor.set(await this.getFileData());
     })();
   }
 
-  private createSaveButton(container: HTMLElement) {
+  private createButtons(container: HTMLElement) {
+    const openButton = document.createElement("button");
+    openButton.innerHTML = "Open Markdown Editor";
```

Comment:
Can we move the buttons directly into the pug file, and reference them using `getElementById` for any click event listeners?  

(I know I started this pattern when I created the first `saveButton` 🤡 this was not very leading by example of me)

Reply:
also think this should be its own ticket, as it will lead to a nontrivial amount of refactoring